File: backend/src/main/java/com/polaris/entity/Tool.java
package com.polaris.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.polaris.common.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 工具实体
 * 对应表：t_tool
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_tool")
public class Tool extends BaseEntity {
    
    /**
     * 分类 ID
     */
    private Long categoryId;
    
    /**
     * 工具名称（英文）
     */
    private String name;
    
    /**
     * 工具名称（中文）
     */
    private String nameZh;
    
    /**
     * 工具描述（英文）
     */
    private String description;
    
    /**
     * 工具描述（中文）
     */
    private String descriptionZh;
    
    /**
     * 图标
     */
    private String icon;
    
    /**
     * URL
     */
    private String url;
    
    /**
     * 颜色类名
     */
    private String colorClass;
    
    /**
     * 背景悬停类名
     */
    private String bgHoverClass;
    
    /**
     * 工具类型
     */
    private Integer toolType;
    
    /**
     * 是否为特色工具：0-否，1-是
     */
    private Integer isFeatured;
    
    /**
     * 浏览次数
     */
    private Long viewCount;
    
    /**
     * 使用次数
     */
    private Long useCount;
    
    /**
     * 排序顺序
     */
    private Integer sortOrder;
    
    /**
     * 状态：0-禁用，1-启用
     */
    private Integer status;
    
    /**
     * 分类名称（英文）- 临时字段，用于关联查询
     */
    @TableField(exist = false)
    private String categoryName;
    
    /**
     * 分类名称（中文）- 临时字段，用于关联查询
     */
    @TableField(exist = false)
    private String categoryNameZh;
    
    /**
     * 最后使用时间 - 临时字段，用于最近使用列表
     */
    @TableField(exist = false)
    private java.time.LocalDateTime lastUsedAt;
}

File: polaris-tools/types/tool.ts
/** 工具与使用记录相关类型 */

export interface Tool {
  id: string;
  numericId?: number;
  title: string;
  title_zh?: string;
  description: string;
  description_zh?: string;
  icon: string;
  category?: string;
  category_zh?: string;
  lastUsed?: string;
  lastUsed_zh?: string;
  path?: string;
  colorClass?: string;
  bgHoverClass?: string;
  accentColor?: string;
}

export interface ToolCreateRequest {
  categoryId: number;
  name: string;
  nameZh?: string;
  description?: string;
  descriptionZh?: string;
  icon: string;
  url?: string;
  colorClass?: string;
  bgHoverClass?: string;
  toolType?: number;
  isFeatured?: number;
  sortOrder?: number;
}

export interface ToolUpdateRequest {
  categoryId?: number;
  name?: string;
  nameZh?: string;
  description?: string;
  descriptionZh?: string;
  icon?: string;
  url?: string;
  colorClass?: string;
  bgHoverClass?: string;
  toolType?: number;
  isFeatured?: number;
  sortOrder?: number;
  status?: number;
}

export interface ToolQueryRequest {
  keyword?: string;
  categoryId?: number;
  toolType?: number;
  isFeatured?: number;
  sortBy?: 'sortOrder' | 'viewCount' | 'useCount' | 'createdAt' | 'rating';
  sortOrder?: 'asc' | 'desc';
  page?: number;
  size?: number;
}

export interface ToolResponse {
  id: number;
  categoryId: number;
  categoryName?: string;
  categoryNameZh?: string;
  name: string;
  nameZh?: string;
  description?: string;
  descriptionZh?: string;
  icon: string;
  url?: string;
  colorClass?: string;
  bgHoverClass?: string;
  toolType: number;
  isFeatured: number;
  viewCount: number;
  useCount: number;
  ratingScore: number;
  ratingCount: number;
  reviewCount: number;
  isFavorited?: boolean;
  sortOrder: number;
  status: number;
  createdAt: string;
  updatedAt: string;
  lastUsedAt?: string;
}

export interface ToolUsageResponse {
  id: number;
  toolId: number;
  toolName: string;
  toolNameZh?: string;
  toolIcon: string;
  usedAt: string;
  duration?: number;
}
